src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```diff
@@ -422,4 +427,50 @@ public String maskEmail(String emailToMask) {
         }
         return emailToMask;
     }
+
+    public String getMiData() {
+        List<String> returnedData = artefactRepository.getMiData();
+        StringBuilder builder = new StringBuilder(146);
+        builder.append("artefact_id,display_from,display_to,language,provenance,sensitivity,"
+                           + "source_artefact_id,type,content_date,court_id,court_name,search\n");
+        for (String s : returnedData) {
+            String[] splitString = s.split(",", 12);
+            long one = 1;
+            builder.append(Arrays.stream(splitString).limit(splitString.length - one)
+                               .collect(Collectors.joining(","))).append(',');
+            try {
+                builder.append(jsonDestroyer(splitString[splitString.length - 1]));
+            } catch (Exception e) {
+                log.error(e.getMessage());
+                builder.append("JSON Error\n");
+            }
+        }
+        return builder.toString();
+    }
+
+    private String jsonDestroyer(String json) throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
If we have thousands of records, the jsonDestroyer is going to be called every time and each time we are creating a new instance of ObjectMapper. It might be better to create a static ObjectMapper at the beginning of this class and just reuse the same one

Reply:
Ooohhh this is a great idea. Thanks!